### ISSUE TITLE
Add feature to append instead of overwrite with cy.writeFile

### ISSUE THREAD

**Issue by cameronc56** [cypress-io/cypress#1249]
- Operating System:
- Cypress Version:
- Browser Version:

### Is this a Feature or Bug?
Feature Request

### Current behavior:
`cy.writeFile` will only overwrite. In order to append, you have to read a file (making sure it already exists...), add the contents of that file to what you want to append to it in memory, and then do another writeFile call. 

### Desired behavior:
add a `{"append": true}` option to `cy.writeFile`.

Since I am dynamically generating tests, this would allow me to aggregate specific metrics from each test into a single JSON file without working around it.

**Comment by jennifer-shehane** [comment]
Relevant code:
- https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/files.coffee#L66
- https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/files.coffee#L25

Also lots of tests for this command that would need updating.

**Comment by maguijo** [comment]
So...having just started with Cypress a few hours ago, I could be very wrong about this. However for what it's worth, this is how I solved the problem:
I installed npm package cvsdata https://www.npmjs.com/package/csvdata
I used the --save-dev flag
then in the plugins index.js file I put this code: 
`const csvdata = require('csvdata') `
`  on('task', {`
   `   log (message) {`
      `  csvdata.write('./logs.csv', message, {append: true, header: 'stats,numusers'})`
      `  return null`
  `  }`
`  })`

Then, in my simple_spec.js file I fed it some data:
`describe('My First Test', function() {`
`    it('Visits the Kitchen Sink', function() {`
`      cy.visit('https://example.cypress.io')`
`      cy.contains('type').then(($el) => {`
`          var data = "inService, 30";`
`          cy.task('log', data)`
`      })`
`    })`
`})`

And voila. A csv file that appends data each time the test is run.

**Comment by jennifer-shehane** [comment]
@maguijo Cool workaround! Will still leave open as a feature to add support for the opts within the command itself.

**Comment by shaunswales** [comment]
Just wondering why when I try and include the following I get the error message below:

Uncaught TypeError: Cannot read property 'write' of undefined.

Any ideas?

**Comment by maguijo** [comment]
Well, undefined can't have any properties so start there. Can you figure out why csvdata (or whatever variable you assigned to require('csvdata') is undefined?

**Comment by brian-mann** [comment]
@shaunswales you included what specifically you never provided what you included?

**Comment by actionJacksonfv** [comment]
@jennifer-shehane was this expected to be released? Is there a reason why it wasn't?

**Comment by chrisbreiding** [comment]
@actionJacksonfv This feature was released in [v3.1.1](https://docs.cypress.io/guides/references/changelog.html#3-1-1).

**Comment by jennifer-shehane** [comment]
Append contents to the end of a file https://on.cypress.io/writefile#Append-contents-to-the-end-of-a-file

```js
cy.writeFile('path/to/message.txt', 'Hello World', { flag: 'a+' })
```
